pkg/client/thrift_generic.go

```diff
@@ -0,0 +1,101 @@
+package client
+
+import (
+	"context"
+
+	"github.com/cloudwego/kitex/client"
+	"github.com/cloudwego/kitex/client/callopt"
+	"github.com/cloudwego/kitex/client/genericclient"
+	"github.com/cloudwego/kitex/pkg/generic"
+	"github.com/cloudwego/kitex/transport"
+	"github.com/kitex-contrib/kitexcall/pkg/config"
+	"github.com/kitex-contrib/kitexcall/pkg/log"
+)
+
+type ThriftGeneric struct {
+	Provider generic.DescriptorProvider
+	Client   genericclient.Client
+	Generic  generic.Generic
+	conf     *config.Config
+	Resp     interface{}
+}
+
+func NewThriftGeneric() Client {
+	return &ThriftGeneric{}
+}
+
+func (c *ThriftGeneric) Init(conf *config.Config) error {
+	// Waiting for server reflection
+	p, err := generic.NewThriftFileProvider(conf.IDLPath)
+	if err != nil {
+		return err
+	}
+	c.Provider = p
+	c.conf = conf
+
+	g, err := generic.JSONThriftGeneric(p)
+	if err != nil {
+		return err
+	}
+	c.Generic = g
+
+	opts, err := c.BuildClientOptions()
+	if err != nil {
+		return err
+	}
+
+	cli, err := genericclient.NewClient(conf.Service, c.Generic, opts...)
+	if err != nil {
+		return err
+	}
+	c.Client = cli
+	return nil
+}
+
+func (c *ThriftGeneric) Call() error {
+	opts, err := c.BuildCallOptions()
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.Client.GenericCall(context.Background(), c.conf.Method, c.conf.Data, opts...)
+	if err != nil {
+		return err
+	}
+
+	c.Resp = resp
+	return nil
+}
+
+func (c *ThriftGeneric) Output() error {
+	result, err := log.Format(log.JF, c.Resp.(string))
+	if err != nil {
+		return err
+	}
+
+	log.Success(result)
+	return nil
+}
+
+func (c *ThriftGeneric) BuildClientOptions() ([]client.Option, error) {
+	var opts []client.Option
+	opts = append(opts, client.WithHostPorts(c.conf.Endpoint...))
+	if c.conf.Transport != "" {
```

Comment:
Do note that `TTHeader` is necessary for passing through metainfo.

Reply:
I will consider this, metainfo has not been added yet.